docs/source/install.rst

```diff
@@ -173,3 +221,40 @@ Building and installing UCX-Py can be done via `pip install`. For example:
     pip install -v .
     # or for develop build
     pip install -v -e .
+
+
+UCX-Py with custom UCX install
```

Comment:
This whole section feels very complicated to document, and probably indicates that we should improve the UX. I've been working on a more general solution, but in the interim one way to make this more configurable at runtime would be for the `libucx.load_library` function to accept some argument(s) that controls the load order of the library. You could also have that controllable by a generic environment variable, or have ucx-py change the arguments it passes to that function based on an environment variable. That would be one way to force it to prefer using the library within the wheel in a more ergonomic manner than having to specify `LD_LIBRARY_PATH`.

Reply:
I had written something really similar to this but then stopped myself... these docs as written are an accurate description of the current state, and would be regardless of the changes in this PR.

Will defer to you and @pentschev on whether you want to block this PR until that interface is changed or just document that in https://github.com/rapidsai/ucx-wheels/issues as a feature request. I personally can't commit to the design work necessary to make a change like that to `libucx.load_library()` right now, but if you two come to an agreement about how you want it to look I could help with the implementation.

Reply:
I agree it's complicated but we don't have a simpler solution _right now_. James and I discussed that before and reached an agreement in https://github.com/rapidsai/ucx-py/pull/1048#discussion_r1662568912 . While I also agree `libucx.load_library` would be a nice solution, we don't have a true use case for that right now and I don't want that any of us has to spend the time working on a niche problem. If someone really _wants_ to do so in the near term, that would be nice, otherwise keeping things as is should be sufficient for now and thus I vote to leave it as is for this PR.

Reply:
OK got it, that's fine.

Reply:
Actually I'm writing a response to #1047 , and while re-reading the original request the use case is exactly this one. With that said, I think `libucx.load_library()` would be the proper solution for that user's use case and thus there is an actual use case, unlike what I wrote above. However, it's also important to note that someone is going to need to take the time to implement that which I know is not any of us have lots of these days. Therefore, I'm thinking of replying with the up-to-date instructions (using `LD_LIBRARY_PATH`) and if the user requires something more sophisticated we should try and engage efforts from his side to get that done sooner as that is the only known use case right now. WDYT @jameslamb @vyasr ?